docs/troubleshooting/kubernetes.rst

```diff
@@ -1,13 +1,223 @@
+
+   apiVersion: extensions/v1beta1
+   kind: Deployment
+   metadata:
+     name: k8s-bigip-ctlr-deployment
+     namespace: kube-system
+   ...
+             args: [
+               "--bigip-username=$(BIGIP_USERNAME)",
+               "--bigip-password=$(BIGIP_PASSWORD)",
+               "--bigip-url=10.190.24.171",
+               "--bigip-partition=kubernetes",
+               "--namespace=<my-namespace>",
+               ]
+   ...
+
+
+Are the Service name and port provided in your virtual server ConfigMap correct?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. code-block:: yaml
+   :caption: Sample Kubernetes Service
+   :emphasize-lines: 4, 11
+
+   kind: Service
+   apiVersion: v1
+   metadata:
+     name: hello
+   spec:
+     selector:
+       app: hello
+       tier: backend
+     ports:
+     - protocol: TCP
+       port: 80
+       targetPort: http
+
+*Source:* `Connect a Front End to a Back End Using a Service <https://kubernetes.io/docs/tasks/access-application-cluster/connecting-frontend-backend/#creating-the-backend-service-object>`_
+
+.. code-block:: yaml
+   :caption: Excerpt from a sample virtual server ConfigMap
+   :emphasize-lines: 10-11
+
+   kind: ConfigMap
+   apiVersion: v1
+   ...
+   data:
+    schema: "f5schemadb://bigip-virtual-server_v0.1.4.json"
+     data: |
+       {
+         "virtualServer": {
+           "backend": {
+             "servicePort": 80,
+             "serviceName": "hello",
+           },
+      ...
+       }
+
+Does the `service type`_ provided in the BIG-IP Controller Deployment match the service type defined for the Service?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+- The default service type used by Kubernetes is ``clusterIP``.
+- The default service type used by the |kctlr| is ``nodeport``.
+
+**If you didn't specify a service type** in the Service definition or the |kctlr| Deployment, you probably have a service type mismatch. See :ref:`Nodeport mode vs Cluster mode <kctlr modes>` for more information about each service type and recommended use.
+
+
+Did you provide valid JSON?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The |kctlr| can only parse valid JSON. Run your desired configurations through a JSON linter before use to avoid potential object creation errors.
+
+
+Have you used the correct version of the `F5 schema`_ to support your version of the BIG-IP Controller?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Additions to the F5 schema made with each version release support the new features in that specific version. For example, if you use v1.3.0 of the Controller with v0.1.2 of the schema, the Controller's core functionality would be fine. You wouldn't, however, be able to use `v1.3 features`_ like DNS resolution or :ref:`adding health monitors for OpenShift Routes <add health monitor to route>`.
```

Comment:
Do we have a chart mapping the versions? That would be nice to include if possible.

Reply:
@f5yacobucci - I made a similar comment on this and suggested that @jputrino link to that schema versioning table.

Reply:
Also an @rsalsa duplicate

Reply:
Yes. 
http://clouddocs.f5.com/containers/v2/releases_and_versioning.html#f5-schema